src/aleph/http/client.clj

```diff
@@ -554,8 +554,11 @@
                     ;; might be different in case we use :multipart
                     (reset! save-body body))
 
-                  (netty/safe-execute ch
-                    (http/send-message ch true ssl? req' body))))
+                  (d/catch'
+                   (netty/safe-execute ch
+                                       (http/send-message ch true ssl? req' body))
+                   (fn [^Throwable e]
+                     (s/put! responses (d/error-deferred e))))))
```

Comment:
Better to thread this with `->` so it more closely looks like standard try/catch